int16 position1
int16 position2
